--- origsrc/libvolume_id-0.81.0/Makefile	2007-12-26 18:52:36.000000000 -0600
+++ src/libvolume_id-0.81.0/Makefile	2008-01-24 20:35:08.227250000 -0600
@@ -15,7 +15,7 @@
 SHLIB_CUR = 0
 SHLIB_REV = 81
 SHLIB_AGE = 0
-SHLIB = libvolume_id.so.$(SHLIB_CUR)
+SHLIB = cygvolume_id-$(SHLIB_CUR).dll
 
 OBJS=				\
 	ext.o			\
@@ -72,7 +72,7 @@
 .shlib/%.o: %.c
 	$(E) "  CC      " $@
 	$(Q) mkdir -p $(dir $@)
-	$(Q) $(CC) -c $(CFLAGS) -fPIC $< -o $@
+	$(Q) $(CC) -c $(CFLAGS) -DPIC $< -o $@
 
 libvolume_id.a: $(HEADERS) $(OBJS)
 	$(Q) rm -f $@
@@ -84,7 +84,7 @@
 $(SHLIB): $(HEADERS) exported_symbols $(addprefix .shlib/,$(OBJS))
 	$(E) "  CC      " $@
 	$(Q) $(CC) -shared $(CFLAGS) -o $@ \
-		-Wl,-soname,libvolume_id.so.$(SHLIB_CUR),--version-script,exported_symbols \
+		-Wl,--enable-auto-image-base,--out-implib,libvolume_id.dll.a \
 		$(addprefix .shlib/,$(OBJS))
 
 libvolume_id.pc:
